worker/uniter/runner/jujuc/network-get_test.go

```diff
@@ -62,32 +112,72 @@ func (s *NetworkGetSuite) TestNetworkGet(c *gc.C) {
 		args:    []string{""},
 		out:     `no binding name specified`,
 	}, {
-		summary: "binding name given, no --primary-address given",
-		code:    2,
-		args:    []string{"foo"},
-		out:     `--primary-address is currently required`,
+		summary: "unknown binding given, no --primary-address given",
+		code:    1,
+		args:    []string{"unknown"},
+		out:     `no network config found for binding "unknown"`,
 	}, {
 		summary: "unknown binding given, with --primary-address",
 		args:    []string{"unknown", "--primary-address"},
 		code:    1,
-		out:     "insert server error for unknown binding here",
+		out:     `no network config found for binding "unknown"`,
 	}, {
-		summary: "valid arguments, API server returns no config",
+		summary: "API server returns no config for this binding, with --primary-address",
 		args:    []string{"valid-no-config", "--primary-address"},
 		code:    1,
 		out:     `no network config found for binding "valid-no-config"`,
 	}, {
+		summary: "API server returns no config for this binding, no --primary-address",
+		args:    []string{"valid-no-config"},
+		code:    1,
+		out:     `no network config found for binding "valid-no-config"`,
+	}, {
 		summary: "explicitly bound, extra-binding name given with --primary-address",
 		args:    []string{"known-extra", "--primary-address"},
 		out:     "10.20.1.42",
 	}, {
+		summary: "explicitly bound, extra-binding name given without --primary-address",
+		args:    []string{"known-extra"},
+		out: `- macaddress: "00:11:22:33:44:22"
+  interfacename: eth2
+  addresses:
+  - address: 10.20.1.42
+    cidr: 10.20.1.42/24
+  - address: fc00::1
+    cidr: fc00::/64`,
+	}, {
 		summary: "explicitly bound relation name given with --primary-address",
 		args:    []string{"known-relation", "--primary-address"},
 		out:     "10.10.0.23",
 	}, {
+		summary: "explicitly bound relation name given without --primary-address",
+		args:    []string{"known-relation"},
+		out: `- macaddress: "00:11:22:33:44:00"
+  interfacename: eth0
+  addresses:
+  - address: 10.10.0.23
+    cidr: 10.10.0.0/24
+  - address: 192.168.1.111
+    cidr: 192.168.1.0/24
+- macaddress: "00:11:22:33:44:11"
+  interfacename: eth1
+  addresses:
+  - address: 10.10.1.23
+    cidr: 10.10.1.0/24
+  - address: 192.168.2.111
+    cidr: 192.168.2.0/24`,
+	}, {
 		summary: "implicitly bound binding name given with --primary-address",
 		args:    []string{"known-unbound", "--primary-address"},
-		out:     "10.33.1.8", // preferred private address used for unspecified bindings.
+		out:     "10.33.1.8",
```

Comment:
'implicitly bound binding'? I wonder if this would be clearer as
 'no user requested binding falls back to the machines private address'